c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,101 @@
-﻿using Newtonsoft.Json;
+﻿using JsonDiffPatchDotNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace c_sharp_json_diff
 {
-    public class Result {
-        public List<Diff> Diffs { get; set; } = new List<Diff>();
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        private string _leftJsonString = "";
+        private string _rightJsonString = "";
+
+        public JsonDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            _leftJsonString = left;
+            _rightJsonString = right;
+        }
+
+        public JObject NumericDiff(bool isAbsoluteDiff)
```

Comment:
Flip the parameters from the constructor and this function. Move the `isAbsoluteDiff` to the class level, and move the left/right variables to the function.

The client code can then use the same instance to map multiple left/right combinations with the same behaviour for absolute diffs.

Reply:
You can also add a default constructor that uses default values for any configuration options.